src/web/components/CMP.tsx

```diff
@@ -1,33 +1,52 @@
-import React, { useState, useEffect } from 'react';
-import {
-    shouldShow,
-    setErrorHandler,
-} from '@guardian/consent-management-platform';
-import { ConsentManagementPlatform } from '@guardian/consent-management-platform/lib/ConsentManagementPlatform';
+import React, { useState, useEffect, Suspense } from 'react';
+
+import { initPerf } from '@root/src/web/browser/initPerf';
+
+const ConsentManagementPlatform = React.lazy(() => {
+    const { start, end } = initPerf('ConsentManagementPlatform');
+    start();
+    return import(
+        /* webpackChunkName: "ConsentManagementPlatform" */ '@guardian/consent-management-platform/lib/ConsentManagementPlatform'
+    ).then(module => {
+        end();
+        return { default: module.ConsentManagementPlatform };
+    });
+});
 
 export const CMP = () => {
     const [show, setShow] = useState(false);
 
     useEffect(() => {
-        if (shouldShow()) {
-            setShow(true);
+        const { start, end } = initPerf(
+            'consent-management-platform-utilities',
+        );
+        start();
+        import(
+            /* webpackChunkName: "consent-management-platform-utilities" */ '@guardian/consent-management-platform'
+        ).then(({ shouldShow, setErrorHandler }) => {
+            end();
+            if (shouldShow()) {
+                setShow(true);
 
-            // setErrorHandler takes function to be called on errors in the CMP UI
-            setErrorHandler((errMsg: string): void => {
-                const err = new Error(errMsg);
+                // setErrorHandler takes function to be called on errors in the CMP UI
+                setErrorHandler((errMsg: string): void => {
+                    const err = new Error(errMsg);
 
-                window.guardian.modules.sentry.reportError(err, 'cmp');
-            });
-        }
+                    window.guardian.modules.sentry.reportError(err, 'cmp');
+                });
+            }
+        });
     }, []);
 
     return (
         <>
             {show && (
-                <ConsentManagementPlatform
-                    source="dcr"
-                    onClose={() => setShow(false)}
-                />
+                <Suspense fallback={<></>}>
```

Comment:
_bites nails_

Reply:
fallback is required, does not support `null`, but we should not render anything while CMP is loading, I could write it as `React.Fragment`.

btw this PR is more about dynamic imports of, this change is from the base PR https://github.com/guardian/dotcom-rendering/pull/1376

Reply:
Because these changes are based on changes made in another branch  may I suggest editing the branch you're merging into on this PR to be that base branch? Only temporarily until it gets merged. That way you'll only see the relevant changes in this PR, which makes it easier to read.
Examples: https://github.com/guardian/frontend/pull/22269 or https://github.com/guardian/frontend/pull/22268